build,sql,cli: reveal the CockroachDB version number

knz opened the pull request:
Previous to this patch, the vtable `crdb_internal.node_build_info`,
the built-in SQL function `version()` and the command-line utilities
would only print the build tag as a version number.

This patch augments this display by prefixing it by the
`VersionPrefix()`, computed after the current branch.

Before:

```
CockroachDB CCL 9fcb2f35e (freebsd amd64, built 2017/09/18 18:27:19, go1.9)
```

After:

```
CockroachDB CCL v1.1-9fcb2f35e (freebsd amd64, built 2017/09/18 18:27:19, go1.9)
```

cc @jordanlewis @justinj

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/18556)
<!-- Reviewable:end -->

benesch reviewed (approved):
LGTM, but is `version()` subject to backwards-compatibility requirements?

knz commented:
The overall format of version() doesn't change. From the client perspective where the VersionPrefix now appears would have been an opaque tag of variable length previously. clients that assume an opaque tag will continue to work.

knz committed:
build,sql,cli: reveal the CockroachDB version number

Previous to this patch, the vtable `crdb_internal.node_build_info`,
the built-in SQL function `version()` and the command-line utilities
would only print the build tag as a version number.

This patch augments this display by prefixing it by the
`VersionPrefix()`, computed after the current branch.

Before:

```
CockroachDB CCL 9fcb2f35e (freebsd amd64, built 2017/09/18 18:27:19, go1.9)
```

After:

```
CockroachDB CCL v1.1-9fcb2f35e (freebsd amd64, built 2017/09/18 18:27:19, go1.9)
```

benesch commented:
But clients that assumed that token was a SHA would break with this change,
right? I suspect this doesn't much matter, but wanted to flag it.

On Mon, Sep 18, 2017 at 2:47 PM, kena <notifications@github.com> wrote:

> The overall format of version() doesn't change. From the client
> perspective where the VersionPrefix now appears would have been an opaque
> tag of variable length previously. clients that assume an opaque tag will
> continue to work.
>
> —
> You are receiving this because your review was requested.
> Reply to this email directly, view it on GitHub
> <https://github.com/cockroachdb/cockroach/pull/18556#issuecomment-330318672>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AA15IECQ33KV-7FxtV47ZI4LwqAmXCxvks5sjrrbgaJpZM4PbVd4>
> .
>

dt commented:
clients already couldn't assume it was a SHA, but they generally *could* assume it was a meaningful scm ref, which they can't with this... but I don't think that's a big deal. this output is more useful to humans anyway, and we can fix the machines if they care.

knz commented:
Thank you!
